adobe_vipm/flows/mpt.py

```diff
@@ -1,6 +1,90 @@
+import copy
+import json
+from functools import wraps
+
+from adobe_vipm.flows.constants import STRUCTURED_PARAMETERS
 from adobe_vipm.flows.errors import wrap_http_error
 
 
+def unpack_structured_parameters(parameters):
+    """
+    Temporary solution that unpacks structured values of complex parameters (Address, Contact)
+    from json string to usual dictionary
+    """
+
+    def _unpack(parameter):
+        if parameter["type"] in STRUCTURED_PARAMETERS:
+            parameter = copy.deepcopy(parameter)
+            parameter["value"] = json.loads(parameter["value"]) if parameter["value"] else None
+        return parameter
+
+    if "fulfillment" in parameters:
+        parameters["fulfillment"] = [_unpack(p) for p in (parameters["fulfillment"] or [])]
+    if "ordering" in parameters:
+        parameters["ordering"] = [_unpack(p) for p in (parameters["ordering"] or [])]
+
+    return parameters
+
+
+def pack_structured_parameters(parameters):
+    """
+    Temporary solution that packs structured values of complex parameters (Address, Contact)
+    from dictionary to json string
+    """
+
+    def _pack(parameter):
+        if parameter["type"] in STRUCTURED_PARAMETERS:
+            parameter = copy.deepcopy(parameter)
+            parameter["value"] = json.dumps(parameter["value"]) if parameter["value"] else None
+        return parameter
+
+    if "fulfillment" in parameters:
+        parameters["fulfillment"] = [_pack(p) for p in (parameters["fulfillment"] or [])]
+    if "ordering" in parameters:
+        parameters["ordering"] = [_pack(p) for p in (parameters["ordering"] or [])]
+
+    return parameters
+
+
+def pack_decorator(parameters_kwarg_name):
+    def _decorator(f):
+        @wraps(f)
+        def _wrapper(*args, **kwargs):
+            if parameters_kwarg_name in kwargs:
+                kwargs[parameters_kwarg_name] = pack_structured_parameters(
+                    kwargs[parameters_kwarg_name]
+                )
+            return f(*args, **kwargs)
+
+        return _wrapper
+
+    return _decorator
+
+
+def unpack_decorator(return_entity_property_name):
+    def _decorator(f):
+        @wraps(f)
+        def _wrapper(*args, **kwargs):
+            entity = f(*args, **kwargs)
+            entity[return_entity_property_name] = unpack_structured_parameters(
+                entity[return_entity_property_name],
+            )
+
+            return entity
+
+        return _wrapper
+
+    return _decorator
+
+
+@unpack_decorator("parameters")
```

Comment:
Since you have to add the logic to pack inside the function I would prefer also to call the unpack inside the function instead of using the decorator because IMO is more clear. Anyway up to you, it's a temporary solution.

Reply:
yup, in order not to change functions I added decorators to pack/unpack. Easy to remove ;-)